skglm/datafits/single_task.py

```diff
@@ -467,3 +467,58 @@ def gradient_scalar_sparse(self, X_data, X_indptr, X_indices, y, Xw, j):
 
     def intercept_update_self(self, y, Xw):
         pass
+
+
+class Gamma(BaseDatafit):
+    r"""Gamma datafit.
+
+    The datafit reads::
+
+    (1 / n_samples) * \sum_i (Xw_i + y_i * exp(-Xw_i))
+
+    Note:
+    ----
+    The class is jit compiled at fit time using Numba compiler.
+    This allows for faster computations.
+    """
+
+    def __init__(self):
+        pass
+
+    def get_spec(self):
+        pass
+
+    def params_to_dict(self):
+        return dict()
+
+    def initialize(self, X, y):
+        if (y <= 0).sum() > 0:
+            raise ValueError(
+                    "Target vector `y` should only take positive values " +
+                    "when fitting a Gamma model.")
+
+    def initialize_sparse(self, X_data, X_indptr, X_indices, y):
+        if (y <= 0).sum() > 0:
+            raise ValueError(
+                    "Target vector `y` should only take positive values " +
+                    "when fitting a Gamma model.")
+
+    def raw_grad(self, y, Xw):
+        """Compute gradient of datafit w.r.t. ``Xw``."""
+        return (1 - y * np.exp(-Xw)) / len(y)
+
+    def raw_hessian(self, y, Xw):
+        """Compute Hessian of datafit w.r.t. ``Xw``."""
+        return (y * np.exp(-Xw)) / len(y)
+
+    def value(self, y, w, Xw):
+        return (np.sum(Xw + y * np.exp(-Xw)) - 1) / len(y)
```

Comment:
A ``log(y_i)`` is missing, right? 
cf. https://github.com/scikit-learn-contrib/skglm/pull/113/files#r1007810345

```suggestion
        return (np.sum(Xw + y * np.exp(-Xw)) - 1 - np.log(y)) / len(y)
```